ui/pages/send/gas-display/gas-display.js

```diff
@@ -0,0 +1,337 @@
+      ? primaryTotalTextOverrideMaxAmount
+      : secondaryTotalTextOverride;
+  };
+  const renderTotalDetailTotal = () => {
+    if (
+      primaryTotalTextOverride === undefined &&
+      secondaryTotalTextOverride === undefined &&
+      draftTransaction.asset.type === 'NATIVE'
+    ) {
+      return (
+        <div className="confirm-page-container-content__total-value">
+          <LoadingHeartBeat
+            estimateUsed={draftTransaction?.userFeeLevel ?? estimateUsed}
+          />
+          <UserPreferencedCurrencyDisplay
+            type={PRIMARY}
+            key="total-detail-value"
+            value={hexTransactionTotal}
+            hideLabel={!useNativeCurrencyAsPrimaryCurrency}
+          />
+        </div>
+      );
+    }
+    return useNativeCurrencyAsPrimaryCurrency
+      ? primaryTotalTextOverrideMaxAmount
+      : secondaryTotalTextOverride;
+  };
+  const renderTotalDetailText = () => {
+    if (
+      primaryTotalTextOverride === undefined &&
+      secondaryTotalTextOverride === undefined
+    ) {
+      return (
+        <div className="confirm-page-container-content__total-value">
+          <LoadingHeartBeat
+            estimateUsed={draftTransaction?.userFeeLevel ?? estimateUsed}
+          />
+          <UserPreferencedCurrencyDisplay
+            type={SECONDARY}
+            key="total-detail-text"
+            value={hexTransactionTotal}
+            hideLabel={Boolean(useNativeCurrencyAsPrimaryCurrency)}
+          />
+        </div>
+      );
+    }
+    return useNativeCurrencyAsPrimaryCurrency
+      ? secondaryTotalTextOverride
+      : primaryTotalTextOverride;
+  };
+  const renderGasDetailsItem = () => {
+    return (
+      <div className="transaction-item">
+        <TransactionDetailItem
+          key="gas-item"
+          detailTitle={
+            <>
+              {t('transactionDetailGasHeading')}
+              <InfoTooltip
+                contentText={
+                  <>
+                    <p>
+                      {t('transactionDetailGasTooltipIntro', [
+                        isMainnet ? t('networkNameEthereum') : '',
+                      ])}
+                    </p>
+                    <p>{t('transactionDetailGasTooltipExplanation')}</p>
+                    <p>
+                      <a
+                        href="https://community.metamask.io/t/what-is-gas-why-do-transactions-take-so-long/3172"
+                        target="_blank"
+                        rel="noopener noreferrer"
+                      >
+                        {t('transactionDetailGasTooltipConversion')}
+                      </a>
+                    </p>
+                  </>
+                }
+                position="top"
+              >
+                <i className="fa fa-info-circle" />
+              </InfoTooltip>
+            </>
+          }
+          detailText={
+            <div className="confirm-page-container-content__currency-container test">
```

Comment:
I can't say that I like using confirm-page styles for this component, its muddling the scss and using styles from another part of the codebase... also why the 'test' class? 

My preference would be to copy the pertinent styles into this gas-display component (replacing as much of them with the Box component as possible) .. this might be duplicative but it makes this component more  resilient to changes in the other component.

Reply:
@brad-decker Done [here](https://github.com/MetaMask/metamask-extension/pull/14634/commits/30e2c90ae66cf2a12aeb11e5ea35daa7cfd60340).